### PR TITLE
Story 36: Admin Invoice Show Page: Update Invoice Status

### LINKED ISSUE #79: 36. Admin Invoice Show Page: Update Invoice Status
As an admin     
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
I see the invoice status is a select field
And I see that the invoice's current status is selected
When I click this select field,
Then I can select a new status for the Invoice,
And next to the select field I see a button to "Update Invoice Status"
When I click this button
I am taken back to the admin invoice show page
And I see that my Invoice's status has now been updated

### PR DESCRIPTION
**Pull request by JorjaF**
__1__ Wrote Tests ____ Implemented ____ Reviewed

Necessary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes: this completes story 36 the update admin invoice status (I did a little refactor on the show page so that it matches the wire frame better)

    description closes #79 

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [x] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(REQUIRED)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: app/views/admin/invoices/show.html.erb [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
 
+  <%= form_with(model: @invoice, url: admin_invoice_path(@invoice), method: :patch) do |form| %>
+    <div>
+      <label for="invoice_status">Invoice Status:</label>
+      <%= form.select :status, options_for_select(Invoice.statuses.keys.map { |status| [status.titleize, status]}, @invoice.status) %>
+      <%= form.submit 'Update Invoice Status' %>
+    </div>
+  <% end %>
+
+  <p>Created at: <%= @invoice.created_at.strftime('%A, %B %d, %Y') %></p>
```
Comment (z-fitch) [inline app/views/admin/invoices/show.html.erb]: This might be too much logic for the view

**Review by z-fitch — CHANGES_REQUESTED** [review]
The time format method needs to be moved out of the view

File: app/views/admin/invoices/show.html.erb (hunk 1 above)
Comment (JorjaF) [reply app/views/admin/invoices/show.html.erb]: I did not think of that. Per your suggestion I moved it over to the model.

File: app/models/invoice.rb [hunk 2]
```diff
@@ -22,4 +22,8 @@ def total_revenue
   def total_revenue_to_currency
     ActiveSupport::NumberHelper::number_to_currency(total_revenue.to_f / 100)
   end
+
+  def self.status_list_for_select_menu
+    statuses.keys.map { |status| [status.titleize, status]}
```
Comment (Daniel-Gallagher92) [inline app/models/invoice.rb:28]: Is this Ruby? / Are we allowed to use Ruby on this project?

File: app/controllers/admin/invoices_controller.rb [hunk 3]
```diff
@@ -7,4 +7,16 @@ def show
   def index
     @invoices = Invoice.all
   end
+
+  def update
+    @invoice = Invoice.find(params[:id])
+    @invoice.update(invoice_params)
+    redirect_to admin_invoice_path(@invoice)
```
Comment (ethanrossblack) [inline app/controllers/admin/invoices_controller.rb:14]: nice advanced routing redirecting

File: app/views/admin/invoices/show.html.erb (hunk 1 above)
Comment (ethanrossblack) [reply app/views/admin/invoices/show.html.erb]: I might push that change of date logic back up to the PR. I'm not seeing it in the model or the view.

File: spec/features/admin/admin_invoice_show_spec.rb [hunk 4]
```diff
@@ -39,7 +39,7 @@
 
     expect(current_path).to eq(admin_invoice_path(@invoice_1))
     expect(page).to have_content(@invoice_1.id)
-    expect(page).to have_content(@invoice_1.status)
+    expect(page).to have_content(@invoice_1.status.to_s.capitalize)
```
Comment (ethanrossblack) [inline spec/features/admin/admin_invoice_show_spec.rb]: With the enum, do we need to turn it back into a string?

File: app/views/admin/invoices/show.html.erb (hunk 1 above)
Comment (ethanrossblack) [reply app/views/admin/invoices/show.html.erb]: ooooooh, wait, I made an invoice method to that stringifies the date. Call `@invoice.formatted_date`

File: app/controllers/admin/invoices_controller.rb (hunk 3 above)
Comment (JorjaF) [reply app/controllers/admin/invoices_controller.rb:14]: thank you

File: app/models/invoice.rb (hunk 2 above)
Comment (JorjaF) [reply app/models/invoice.rb:28]: we got the okay from chris on this and I added the commit from chris

File: app/views/admin/invoices/show.html.erb (hunk 1 above)
Comment (JorjaF) [reply app/views/admin/invoices/show.html.erb]: Brilliant that worked like a charm and I updated this

File: spec/features/admin/admin_invoice_show_spec.rb (hunk 4 above)
Comment (JorjaF) [reply spec/features/admin/admin_invoice_show_spec.rb]: I took it out the .to_s

**Review by ethanrossblack — APPROVED** [review]
Looks good to me! Nice work.

**Review by z-fitch — APPROVED** [review]

**Pull request merged**